### PR TITLE
feat(sdk): Expose `prepare_encrypted_file` function

### PR DESCRIPTION
**Pull request by swedgwood**
Exposed a function for uploading encrypted files and preparing an `EncryptedFile`, this would be useful for people who are creating custom events (like me!) and want to leverage the existing `EncryptedFile` definition.

I'm not sure if this is the most sensible place for this function, feel free to suggest somewhere else more suitable if so

### CODE REVIEW

**Review by poljar — CHANGES_REQUESTED** [review]
Thanks for this. I think the places is fine, at least I can't think of any better place.

The docs could handle a bit more verbosity and an example, otherwise looks fine.

**Review by poljar — APPROVED** [review]
Thanks, looks good.

**Comment by poljar** [comment]
Oh, seems that `cargo fmt` isn't happy yet. Could you run `cargo xtask fixup style` and push the resulting changes?

File: crates/matrix-sdk/src/encryption/mod.rs [hunk 1]
```diff
@@ -108,6 +108,66 @@ impl Client {
         Ok(response)
     }
 
+    /// Construct a [`EncryptedFile`][ruma::events::room::EncryptedFile] by encrypting
+    /// and uploading a provided reader.
```
Comment (jplatte) [inline crates/matrix-sdk/src/encryption/mod.rs:112]:
```suggestion
    /// Construct a [`EncryptedFile`][ruma::events::room::EncryptedFile] by
    /// encrypting and uploading a provided reader.
```

**Comment by jplatte** [comment]
Yes, this will be part of the release but no, there is no ETA for it.

**Pull request merged**